guide/creating-your-bot/event-handling.md

```diff
@@ -82,22 +85,17 @@ for (const file of eventFiles) {
-:::
-
-The `Client` class in discord.js extends the `EventEmitter` class. Therefore, the `client` object also has these `on` and `once` methods that you can use to register events. These methods take two arguments: the name of the event and a callback function.
-
-The callback function passed takes argument(s) returned by its respective event, collects them in an `args` array using the `...` [rest parameter syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters), then calls `event.execute` function while passing in the `args` array using the `...` [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). They are used here because different events in discord.js have different numbers of arguments. The rest parameter collects these variable number of arguments into a single array, and the spread syntax then takes these elements and passes them to the `execute` function.
+The callback function passed takes argument(s) returned by its respective event, collects them in an `args` array using the `...` [rest parameter syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters), then calls `event.execute()` while passing in the `args` array using the `...` [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). They are used here because different events in discord.js have different numbers of arguments. The rest parameter collects these variable number of arguments into a single array, and the spread syntax then takes these elements and passes them to the `execute` function.
 
 After this, listening for other events is as easy as creating a new file in the `events` folder. The event handler will automatically retrieve and register it whenever you restart your bot.
 
 ## Passing `Client` to event files
 
-You may have noticed how important the `Client` class is. You created a `client` instance of this class in the `index.js` file. Most of the time, you can use this `client` instance in other files by either obtaining it from one of the other discord.js structures or function parameters. In your `interactionCreate` event, you can use `interaction.client`. When you don't have access to any of the structures with the `client` property, you'll have to use the latter method. A prime example of this is the `ready` event.
+You may have noticed how important the <DocsLink path="class/Client" /> class is. You created a `client` instance of this class in the `index.js` file. Most of the time, you can use this `client` instance in other files by obtaining it from one of the other discord.js structures. In the `interactionCreate` event, you can use `interaction.client`.
+
+However, not all events will expose a structure with the `.client` property attached. A prime example of this is the `ready` event.
```

Comment:
`ready` emits with `client` since v13 so this could be updated as well. Otherwise good job on the rework